homeassistant/components/apcupsd/sensor.py

```diff
@@ -473,6 +479,14 @@ def _handle_coordinator_update(self) -> None:
     def _update_attrs(self) -> None:
         """Update sensor attributes based on coordinator data."""
         key = self.entity_description.key.upper()
+        # For most sensors the key will always be available for each refresh. However, some sensors
+        # (e.g., "laststest") will only appear after certain event occurs (e.g., a self test is
+        # performed) and may disappear again after certain event. So we mark the state as "unknown"
+        # when it becomes unknown after such events.
+        if key not in self.coordinator.data:
+            self._attr_native_value = STATE_UNKNOWN
```

Comment:
Don't use `STATE_UNKNOWN`. We use `None` to represent unknown state internally.

Reply:
Thanks for the info! PR #124609 created to fix this!